pkg/autoscaler/autoscaler.go

```diff
@@ -254,295 +257,409 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
+				klog.Infof("[%s remaining] At max cluster size and expecting no more %q events; currently have %v, max=%v",
+					remaining(testDuration), clusterAutoscalerScaledUpGroup, v, clusterExpansionSize-1)
+				return v == uint32(clusterExpansionSize-1)
+			}, framework.WaitShort, pollingInterval).Should(BeTrue())
+
+			By("Deleting workload")
+			scaleDownCounter := newScaleDownCounter(eventWatcher, uint32(clusterExpansionSize-1))
+			Expect(deleteObject(workload.Name, cleanupObjects[workload.Name])).Should(Succeed())
+			delete(cleanupObjects, workload.Name)
+			testDuration = time.Now().Add(time.Duration(framework.WaitLong))
+			Eventually(func() bool {
+				v := scaleDownCounter.get()
+				klog.Infof("[%s remaining] Expecting %v %q events; observed %v",
+					remaining(testDuration), clusterExpansionSize-1, clusterAutoscalerScaleDownEmpty, v)
+				return v == uint32(clusterExpansionSize-1)
+			}, framework.WaitLong, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				podList := corev1.PodList{}
+				err = client.List(ctx, &podList, runtimeclient.InNamespace(workload.Namespace))
+				Expect(err).NotTo(HaveOccurred())
+				for i := range podList.Items {
+					if strings.Contains(podList.Items[i].Name, workloadJobName) {
+						klog.Infof("still have workload POD: %q", podList.Items[i].Name)
+						return false
+					}
 				}
+				return true
+			}, framework.WaitMedium, pollingInterval).Should(BeZero())
+
+			// Delete MachineAutoscalers to prevent scaling while we manually
+			// scale-down the recently created MachineSets.
+			for _, ma := range machineAutoscalers {
+				err := deleteObject(ma.Name, ma)
+				Expect(err).NotTo(HaveOccurred())
+				delete(cleanupObjects, ma.Name)
 			}
-			return true
-		}, framework.WaitMedium, pollingInterval).Should(BeZero())
 
-		// Delete MachineAutoscalers to prevent scaling while we manually
-		// scale-down the recently created MachineSets.
-		for _, ma := range machineAutoscalers {
-			err := deleteObject(ma.Name, ma)
-			Expect(err).NotTo(HaveOccurred())
-			delete(cleanupObjects, ma.Name)
-		}
+			// Delete the transient MachinSets.
+			for _, ms := range machineSets {
+				err := deleteObject(ms.Name, ms)
+				Expect(err).NotTo(HaveOccurred())
 
-		// Delete the transient MachinSets.
-		for _, ms := range machineSets {
-			err := deleteObject(ms.Name, ms)
-			Expect(err).NotTo(HaveOccurred())
+				delete(cleanupObjects, ms.Name)
 
-			delete(cleanupObjects, ms.Name)
+				framework.WaitForMachineSetDelete(client, ms)
+			}
 
-			framework.WaitForMachineSetDelete(client, ms)
-		}
+			// explicitly delete the ClusterAutoscaler
+			// this is needed due to the autoscaler tests requiring singleton
+			// deployments of the ClusterAutoscaler.
+			caName := clusterAutoscaler.GetName()
+			Expect(deleteObject(caName, cleanupObjects[caName])).Should(Succeed())
+			delete(cleanupObjects, caName)
+			Eventually(func() bool {
+				By(fmt.Sprintf("Waiting for ClusterAutoscaler to delete"))
```

Comment:
Since this line is changed already, may as well remove the redundant fmt
```suggestion
				By("Waiting for ClusterAutoscaler to delete")
```